components/datadog/agent/kubernetes_helm.go

```diff
@@ -250,11 +258,19 @@ func buildLinuxHelmValues(baseName, agentImagePath, agentImageTag, clusterAgentI
 		containerRegistry = ""
 		imageName = agentImagePath
 	}
+	// Determine leader election mechanism based on kubeVersion
+	kubeThresholdVersion, _ := semver.NewVersion("1.21.0")
+	leaderElectionType := "lease"
+	if kubeVersion.LessThan(kubeThresholdVersion) {
+		leaderElectionType = "configmap"
+	}
+
 	helmValues := HelmValues{
 		"datadog": pulumi.Map{
-			"apiKeyExistingSecret": pulumi.String(baseName + "-datadog-credentials"),
-			"appKeyExistingSecret": pulumi.String(baseName + "-datadog-credentials"),
-			"checksCardinality":    pulumi.String("high"),
+			"apiKeyExistingSecret":   pulumi.String(baseName + "-datadog-credentials"),
+			"appKeyExistingSecret":   pulumi.String(baseName + "-datadog-credentials"),
+			"leaderElectionResource": pulumi.String(leaderElectionType),
```

Comment:
According to the docs, you can set this to an empty string, and the Agent will automatically use leases when available and fall back to ConfigMaps when they’re not.